test/account_test.rb

```diff
@@ -145,6 +145,30 @@ def app
       assert_equal 302, last_response.status
       assert_match %r{/login$}, last_response.headers["Location"]
     end
+
+    it "redirects back to reset page when there's a 422 error" do
+      stub_heroku_api do
+        post("/auth/finish_reset_password/c45685917ef644198a0fececa10d479a") {
+          [422, MultiJson.encode({ message: "Password too short." })]
+        }
+      end
+      post "/account/password/reset/c45685917ef644198a0fececa10d479a",
+        password: "1234567890ab", password_confirmation: "1234567890ab"
+      assert_equal 302, last_response.status
+      assert_match %r{/account/password/reset/c45685917ef644198a0fececa10d479a$}, last_response.headers["Location"]
+    end
+
+    it "redirects back to reset page when there's a 403 error" do
+      stub_heroku_api do
+        post("/auth/finish_reset_password/c45685917ef644198a0fececa10d479a") {
+          [403, MultiJson.encode({ message: "Password too short." })]
```

Comment:
Should we use another example here instead of "Password is too short" so it's more clear how's that different from 422 handler?